lib/events.js

```diff
@@ -437,9 +441,9 @@ EventEmitter.prototype.removeAllListeners =
         this.removeListener(type, listeners);
       } else if (listeners) {
         // LIFO order
-        do {
-          this.removeListener(type, listeners[listeners.length - 1]);
-        } while (listeners[0]);
+        for (i = listeners.length; i-- > 0;) {
```

Comment:
I've use that because the same is done few lines above: https://github.com/nodejs/node/pull/12043/files#diff-71dcd327d0ca067b490b22d677f81966R366.